src/actions/admin.js

```diff
@@ -0,0 +1,52 @@
+import * as api from '../services/api';
+import History from '../history';
+
+import { GET_PROTECTED_DATA, MODLIST_ERROR } from './types';
+
+export const getModlist = () => {
+  return async dispatch => {
+    try {
+      const response = await api.get({ url: 'moderatorlist' });
+      dispatch({ type: GET_PROTECTED_DATA, payload: response.data });
+    } catch (err) {
+      dispatch({ type: MODLIST_ERROR, payload: err });
+      History.push('/moderatorlist');
+    }
```

Comment:
You can rewrite this in way that `try-catch` and `async` can be removed. This is a preferred method for chaining promises.

```js
api.get(...)
  .then((response) => {
    // On success
   ...
  }).catch((error) => {
    // On failure
    ...
  });
```

Reply:
There are a number of similar blocks where this same advice applies.
```js
try {
  const someConst = await asyncMethodCall();
  ...
} catch(err) {
  ...
}
```

Reply:
I think we'll have to agree to disagree here :) async / await is just syntactic sugar on top of promises, personally I'd stick with either Promises or async / await, or at least keep Promise and async / await based code isolated from one another. Unfortunately this means that with async / await, you need to use try - catch blocks to catch exceptions. Some may not like it and some might feel like it makes the code easier to reason about